user-service/src/main/java/uk/gov/gchq/palisade/service/user/config/StdUserCacheWarmerFactory.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.user.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import uk.gov.gchq.palisade.User;
+import uk.gov.gchq.palisade.service.UserCacheWarmerFactory;
+
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+@ConfigurationProperties
+public class StdUserCacheWarmerFactory implements UserCacheWarmerFactory {
+
+    private String userId;
+    private Set<String> auths;
+    private Set<String> roles;
+
+    public StdUserCacheWarmerFactory() {
+        this.userId = "";
+        this.auths = Collections.emptySet();
+        this.roles = Collections.emptySet();
+    }
+
+    public StdUserCacheWarmerFactory(final String userId, final Set<String> roles, final Set<String> auths) {
+        this.userId = userId;
+        this.auths = auths;
+        this.roles = roles;
+    }
+
+    public String getUserId() {
+        return userId;
+    }
+
+    public void setUserId(final String userId) {
+        this.userId = userId;
```

Comment:
setter methods need requireNonNull

Reply:
Used code generator template to update this

Reply:
I think you need to delete the existing setters and getters and regenerate. Certainly mine generate differently to these. You might also want to add mutator methods, given you currently have your setters return the StdUserCacheWarmerFactory object which setters shouldn't.

Reply:
I think these were created using Adam's original templates. I'll generate them with the new templates and add some mutator methods